src/StatCache.cc

```diff
@@ -0,0 +1,65 @@
+/*
+ *  OpenSCAD (www.openscad.org)
+ *  Copyright (C) 2009-2011 Clifford Wolf <clifford@clifford.at> and
+ *                          Marius Kintel <marius@kintel.net>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  As a special exception, you have permission to link this program
+ *  with the CGAL library and distribute executables, as long as you
+ *  follow the requirements of the GNU GPL in regard to all of the
+ *  software in the executable aside from CGAL.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+#include "StatCache.h"
+#include "printutils.h"
+
+#include <sys/stat.h>
+#include <string>
+#include <unordered_map>
+
+typedef std::unordered_map<std::string, struct stat> StatMap;
+
+static StatMap statMap;

```

Comment:
I am assuming the GUI is event driven, not multithreaded. I.e. I am assuming if one window calls handleDependencies() then that function will run to completion and not be pre-empted by another thread even if another window's timer fires.

I have updated my cache code and removed the explicit clear, so the results are now shared between windows.

I think I know why the second window doesn't update: Because my windows are using the same modules the first window will call updateDependencies and that will update all the modules and return true. When the second window calls updateDependencies they will already be updated, so it will return false. Hence the second window doesn't think it needs to redraw.